src/sinks/aws_kinesis_firehose/config.rs

```diff
@@ -179,29 +215,26 @@ impl KinesisFirehoseSinkConfig {
         }
     }
 
-    pub fn create_client(&self, proxy: &ProxyConfig) -> crate::Result<KinesisFirehoseClient> {
-        let region = (&self.region).try_into()?;
-        let tls_settings = MaybeTlsSettings::from(TlsSettings::from_options(&self.tls)?);
-
-        let client = rusoto::client(Some(tls_settings), proxy)?;
-        let creds = self.auth.build(&region, self.assume_role.clone())?;
-
-        let client = rusoto_core::Client::new_with_encoding(creds, client, self.compression.into());
-        Ok(KinesisFirehoseClient::new_with_client(client, region))
+    pub async fn create_client(&self, proxy: &ProxyConfig) -> crate::Result<KinesisFirehoseClient> {
+        create_client::<KinesisFirehoseClientBuilder>(
+            &self.auth,
+            self.region.region(),
+            self.region.endpoint()?,
+            proxy,
+            &self.tls,
+        )
+        .await
     }
 }
 
 #[derive(Clone)]
 pub struct KinesisRetryLogic;
 
 impl RetryLogic for KinesisRetryLogic {
-    type Error = RusotoError<PutRecordBatchError>;
+    type Error = SdkError<PutRecordBatchError>;
     type Response = KinesisResponse;
 
     fn is_retriable_error(&self, error: &Self::Error) -> bool {
-        match error {
-            RusotoError::Service(PutRecordBatchError::ServiceUnavailable(_)) => true,
-            error => rusoto::is_retriable_error(error),
-        }
+        is_retriable_error(error)
```

Comment:
Another case where we should be handling the service specific error, `PutRecordBatchError`, separately?